transport/src/test/java/io/netty/channel/FileChannelBasedSlicedFileRegionTest.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.channel;
+
+import io.netty.util.internal.ThreadLocalRandom;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.channels.Channels;
+import java.nio.channels.WritableByteChannel;
+import java.util.Arrays;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test for {@link FileChannelBasedSlicedFileRegion}.
+ */
+public final class FileChannelBasedSlicedFileRegionTest {
+
+    private static byte[] TEST_DATA = new byte[1024 * 64];
+
+    private static File TEST_FILE;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        ThreadLocalRandom.current().nextBytes(TEST_DATA);
+        TEST_FILE = File.createTempFile("netty-", ".tmp");
+        TEST_FILE.deleteOnExit();
+        FileOutputStream out = new FileOutputStream(TEST_FILE);
+        try {
+            out.write(TEST_DATA);
+        } finally {
+            out.close();
+        }
+    }
+
+    private byte[] transferBytesTo(SliceableFileRegion region, int length) throws IOException {
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        WritableByteChannel ch = Channels.newChannel(bos);
```

Comment:
does `ch` need to be closed?

Reply:
Not really, it just wraps a `ByteArrayOutputStream`. But anyway, no harm to close it. Fixed with a try finally.